apps/sailfish/qml/components/DockedControls.qml

```diff
@@ -121,14 +125,52 @@ DockedPanel {
                 maximumValue: 100
 
                 onValueChanged: {
-                    kodi.volume = value
+                    if (passiveUpdate)
+                        return
+                    requestVolume(value)
+                }
+
+                // sometimes XBMC doesn't report a volumechanged event back after setting it
+                Timer {
+                    id: volumeChangeTimeout
+                    interval: 500
+                    repeat: false
+                    onTriggered: {
+                        // request volume again
+                        volumeSlider.volumeChangeOutstanding = false
+                        volumeSlider.requestVolume(volumeSlider.volumePending)
+                    }
                 }
 
-                Binding {
-                    target: volumeSlider
-                    property: "value"
-                    value: kodi.volume
+                Component.onCompleted: {
+                    kodi.onVolumeChanged.connect(processChangedVolume)
                 }
+
+                // send volume request, but don't flood xbmc
+                function requestVolume(value) {
+                    volumePending = value
+                    if (!volumeChangeOutstanding) {
+                        kodi.volume = value
+                        volumeChangeOutstanding = true
+                        volumeChangeTimeout.start()
+                    }
+                }
+
+                function processChangedVolume(volume) {
+                    // update UI
+                    passiveUpdate = true
+                    volumeSlider.value = volume
+                    passiveUpdate = false
+                    volumeChangeTimeout.stop()
+                    // send last queued volume change
+                    if (volumeChangeOutstanding) {
+                        volumeChangeOutstanding = false
+                        if (volumePending > -1 && volume !== volumePending) {
+                            requestVolume(volumePending)
+                        }
+                    }
+                }
+
```

Comment:
Looking at the docs that would be `onDownChanged` + checking the `down` value if it's not down (anymore).

But I think my C++ based solution might be better, as it works in all apps, and solves the "Kodi doesn't always send the notification" issue. Although my solution would still "spam" Kodi, but it doesn't keep spamming it until you release the slider.